core/trie/bitarray.go

```diff
@@ -0,0 +1,397 @@
+package trie
+
+import (
+	"bytes"
+	"encoding/binary"
+	"math"
+	"math/bits"
+
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+const (
+	maxUint64 = uint64(math.MaxUint64)
+	bits8     = 8
+)
+
+var (
+	maxBitArray   = [4]uint64{math.MaxUint64, math.MaxUint64, math.MaxUint64, math.MaxUint64}
+	emptyBitArray = new(BitArray)
+)
+
+// BitArray is a structure that represents a bit array with length representing the number of used bits.
+// It uses a little endian representation to do bitwise operations of the words efficiently.
```

Comment:
The same couldn't be achieved with big endian?

Reply:
Big endian works too. However, usually when dealing with low level things like bitwise operations, we use little endian format. If you check most libraries like `https://github.com/holiman/uint256` or `math/big`, they use little endian format. The reason being that most CPUs are little endian so it aligns naturally with how they handle the data. It also simplifies bitwise operations.

However, the tradeoff is that it's slightly harder to debug because we are used to big endian format.